Colorspace Management and Distribution (#4195)
ynput/OpenPype: website/docs/project_settings/settings_project_global.md

Projects always use default project values unless they have [project override](../admin_settings#project-overrides) (orage colour). Any changes in default project may affect all existing projects.
:::

## Color Management (ImageIO)

:::info Default OCIO config
OpenPype is distributing own ocio configs. Those can be found in `{openpype install dir}/{version}/vendor/bin/ocioconfig/OpenColorIOConfigs`. Windows example: `C:\Program Files (x86)\OpenPype\3.14.0\vendor\bin\ocioconfig\OpenColorIOConfigs`
:::

### Using OCIO config
Each path input is supporting [anatomy template key formating](../admin_settings_project_anatomy#available-template-keys) and also environment variables. Default distributed value is set to `{OPENPYPE_ROOT}/vendor/bin/ocioconfig/OpenColorIOConfigs/aces_1.2/config.ocio`. If any of host config is enabled and set to any path, global config path is overriden.

> Config path can be defined to particular shot target with following path input `{root[work]}/{project[name]}/{hierarchy}/{asset}/config/aces.ocio`

Reviewer: Does it need to be for each shot? Or can you define it on a sequence level?

> Config path can be defined to particular shot target with following path input `{root[work]}/{project[name]}/{hierarchy}/{asset}/config/aces.ocio`

Author: > Should this maybe additionally mention that when enabled like this EACH shot must have a config/aces.ocio of its own? (Does that make sense in production?)

@BigRoy Yeah, now when you mention it - I believe it will not be really solid feature for some. I presume some sort of falback, or multi path inputs with order should be implemented. I guess best approach on this would be to add all templated paths targeting all hierarchical paretns. So for example. 

1. `{root[work]}/{project[name]}/{hierarchy}/{asset}/config/aces.ocio`
2. `{root[work]}/{project[name]}/{hierarchy}/config/aces.ocio`
3. `{root[work]}/{project[name]}/config/aces.ocio`

Only that 2nd path will be a bit tricky. All parents will need to be iterated for possible paths to be tested.

> Config path can be defined to particular shot target with following path input `{root[work]}/{project[name]}/{hierarchy}/{asset}/config/aces.ocio`

Author: > Does it need to be for each shot? Or can you define it on a sequence level?

@tokejepsen I guess the approach mentioned above would be addressing what you thing, right?

> Config path can be defined to particular shot target with following path input `{root[work]}/{project[name]}/{hierarchy}/{asset}/config/aces.ocio`

Reviewer: > @tokejepsen I guess the approach mentioned above would be addressing what you thing, right?

Yeah, I think going up the hierarchy from the current asset/version/representation would make sense and make it flexible enough for production.

> ### Using File rules
> File rules are inspired by OCIO v2 configuration ([more info here](https://opencolorio.readthedocs.io/en/latest/guides/authoring/rules.html)). Since each rule is having unique name it can be overidden by host activated File rules (example: `project_settings/nuke/imageio/file_rules/rules`). Pattern input is using REGEX expression syntax (try [regexr.com](https://regexr.com/)). Matching rules procedure's intention is to be used during publishing or loading of representation. Since the publishing procedure is run before integrator formates publish template path, make sure the pattern is working or any work render path.

Reviewer 2: ```suggestion
### Using File rules
File rules are inspired by [OCIO v2 configuration]((https://opencolorio.readthedocs.io/en/latest/guides/authoring/rules.html)). Each rule has a unique name which can be overridden by host-specific _File rules_ (example: `project_settings/nuke/imageio/file_rules/rules`). 

The _input pattern_ matching uses REGEX expression syntax (try [regexr.com](https://regexr.com/)). Matching rules procedure's intention is to be used during publishing or loading of representation. Since the publishing procedure is run before integrator formate publish template path, make sure the pattern is working or any work render path.
```

I've restructured this somewhat. But still couldn't make any sense of this line:

> Matching rules procedure's intention is to be used during publishing or loading of representation. Since the publishing procedure is run before integrator formate publish template path, make sure the pattern is working or any work render path.

What? :D

> :::warning Colorspace name input
> Colorspace name value is string input and no validation is run after saving of changes. We recommand to open a config.ocio file and search exact colorspace names for validation.
> :::

Reviewer 2: ```suggestion
:::warning Colorspace name input
The **colorspace name** value is a raw string input and no validation is run after saving project settings. We recommend to open the specified `config.ocio` file and copy pasting the exact colorspace names.
:::
```
